### PR TITLE
Chore/apollon diagram tests

### PR DESCRIPTION
**Pull request by TobiasPr**
### Motivation and Context

Small refactoring of fullscreen query, enter and exit code: move to utils class
Increasing test coverage

### Test Coverage

- app/exercises/modeling/shared:
  - modeling-editor.component.ts: 0% -> 65.93% 
- app/exercises/quiz/manage/apollon-diagrams (directory) 0% -> 84.38%
  - apollon-diagram-create-form.component.ts 0 % -> 90.91%
  - apollon-diagram-detail.component.ts 0% -> 88.61%
  - apollon-diagram-list.component.ts 0% -> 77.14%
- app/exercises/quiz/manage/apollon-diagrams/exercise-generation (directory) 0% -> 66.85%
  - apollon-quiz-exercise-generation.component.ts 0% -> 84%
  - quiz-exercise-generator.ts 0% -> 79.84%

### Steps for Testing
- Code review

### CODE REVIEW

File: src/test/javascript/spec/component/apollon-diagrams/apollon-diagram-detail.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import { JhiLanguageHelper } from 'app/core/language/language.helper';
+import { ApollonDiagramDetailComponent } from 'app/exercises/quiz/manage/apollon-diagrams/apollon-diagram-detail.component';
+import { TranslateService } from '@ngx-translate/core';
+import { MockTranslateService } from '../../helpers/mocks/service/mock-translate.service';
+import { MockRouter } from '../../helpers/mocks/service/mock-route.service';
+import * as testClassDiagram from '../../util/modeling/test-models/class-diagram.json';
+import { UMLModel, Selection } from '@ls1intum/apollon';
+import { ElementRef } from '@angular/core';
+import { Text } from '@ls1intum/apollon/lib/utils/svg/text';
+import { convertRenderedSVGToPNG } from 'app/exercises/quiz/manage/apollon-diagrams/exercise-generation/svg-renderer';
```
Comment (krusche) [inline src/test/javascript/spec/component/apollon-diagrams/apollon-diagram-detail.component.spec.ts:23]: please remove the line, see Github warning

File: src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    it('getCurrentModel', () => {
+        // @ts-ignore
+        fixture.componentInstance.umlModel = testClassDiagram as UMLModel;
+        fixture.detectChanges();
+        fixture.componentInstance.ngAfterViewInit();
+        expect(fixture.componentInstance['apollonEditor']).toBeTruthy();
+
+        // test
+        const model = fixture.componentInstance.getCurrentModel();
+        // TODO: uncomment after deserialization bugfix in Apollon libary
```
Comment (krusche) [inline src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts:133]: which deserialization issue do you mean? can you reference it here?
Is it easy to fix or more complex?

**Review by krusche — CHANGES_REQUESTED** [review]
Code looks good.
Thanks for removing the duplicated code for full screen functionality.

Would be great if you could fix the GitHub warnings and add some more details about the TODO with the deserialization bug

File: src/main/webapp/app/shared/util/fullscreen.util.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    // check if this component is the current fullscreen component for different browser types
+    if (docElement.fullscreenElement !== undefined) {
+        return docElement.fullscreenElement;
+    } else if (docElement.webkitFullscreenElement !== undefined) {
+        return docElement.webkitFullscreenElement;
+    } else if (docElement.mozFullScreenElement !== undefined) {
+        return docElement.mozFullScreenElement;
+    } else if (docElement.msFullscreenElement !== undefined) {
+        return docElement.msFullscreenElement;
+    }
```
Comment (derLalla) [inline src/main/webapp/app/shared/util/fullscreen.util.ts:15]: an alternative return statement of false would be good here, I suppose. 

```suggestion
    }
    return false;
```

File: src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts (hunk 2 above)
Comment (TobiasPr) [reply src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts:133]: https://github.com/ls1intum/Apollon/issues/146 
Not sure about the complexity, I invested a bit of time and located the issue, but did not yet fully understand why it happens. Can post these information in the Apollon ticket

File: src/main/webapp/app/shared/util/fullscreen.util.ts (hunk 3 above)
Comment (TobiasPr) [reply src/main/webapp/app/shared/util/fullscreen.util.ts:15]: Done

File: src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts (hunk 2 above)
Comment (krusche) [inline src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts:133]:
```suggestion
        // TODO: uncomment after deserialization bugfix in Apollon library, see
```

File: src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    it('getCurrentModel', () => {
+        // @ts-ignore
+        fixture.componentInstance.umlModel = testClassDiagram as UMLModel;
+        fixture.detectChanges();
+        fixture.componentInstance.ngAfterViewInit();
+        expect(fixture.componentInstance['apollonEditor']).toBeTruthy();
+
+        // test
+        const model = fixture.componentInstance.getCurrentModel();
```
Comment (krusche) [inline src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts:132]:
```suggestion
        // const model = fixture.componentInstance.getCurrentModel();
```

**Review by krusche — APPROVED** [review]
Code looks good to me know 👍

**Pull request merged**